book/src/workspace-structure.md

```diff
@@ -0,0 +1,123 @@
+# Workspace Structure
+
+All dist projects, regardless of how many packages they may contain, are
+considered workspaces. A minimalist dist project would be a workspace that
+contains a single package.
+
+Dist has two types of configuration:
+
+- **Workspace**:
+- **Package**: 
+
+The vast majority of configuration for dist is workspace configuration. Because
+we understand language-specific config manifests like `Cargo.toml` and
+`package.json`, package configuration ususally does not require any additional
+files.
+
+## Workspace configuration
+
+A workspace file may be:
+
+* `dist-workspace.toml`
+* (deprecated in 0.23.0) `Cargo.toml` (Rust users only, goes in `[workspace.metadata.dist]`)
+
+### Specifying The Members Of Your Workspace
+
+Create a `workspace.members` field in your `dist-workspace.toml` that points to
+an array of strings. Each string is prefaced with a type:
+
+- `npm`: this indicates a package that is specified by a `package.json`
+- `cargo`: this indicates a package (or nested workspace) that is specified by a `Cargo.toml`. You do not need to specify cargo workspace members individually, you can simply refer to the workspace.
+- `dist`: this indicates a package that is specified by a `dist.toml`
+
+For example:
+
+```toml
+[workspace]
+members = [
+  "npm:path/to/npm/packagejson/dir/",
+  "cargo:path/to/workspace/cargotoml/dir/",
+  "dist:path/to/distoml/dir/"
+]
+```
+
+## Package configuration
+
+A package file may be:
+
+* `dist.toml`
+* `dist-workspace.toml`
+* `Cargo.toml` (for a Rust package)
+* `package.json` (for an npm package)
+
+In the case of a `Cargo.toml` and `package.json`, we'll do our best to find basic package
+info like package name, version, repository, binaries among the native language-specific config.
+
+However these files do not natively support dist-specific config, so you may 
+need to place a dist config *next* to them to specify additional dist-specific
+config. If we see the language-specific package file, we will automatically
+check for a neighbouring dist config and merge its contents in. If you wish to
+"shadow" or change a value that is present in your package manifest- you can
```

Comment:
Nitpick - we probably want to pick a single syntax for dashes and stick to it. These aren't rendering the same as each other.